Add URL to reference of Minka paper used in PCA

hamogu opened the pull request:
This PR adds a link to the reference of the Minka paper implemented in the PCA for `method='mle'`. That paper is cited twice: In the method `_asses_dimension` I add a direct link to the exact paper, in the general docstring of the PCA I add a reference to Minka's page with the algorithm, because on that page there is an extended and updated write-up on this method in addition to a copy of the original paper. That write-up is more useful as an introduction to the method, but obviously has different equation numbers etc. thus it is not a good link for the `_asses_dimension` method, which explicitly refers to "equation (31) from the paper".

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.


#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

hamogu committed:
Add URL to reference of Minka paper used in PCA

This PR adds a link to the reference of the Minka paper implemented in the PCA for `method='mle'`. That paper is cited twice: In the method `_asses_dimension` I add a direct link to the exact paper, in the general docstring of the PCA I add a reference to Minka's page with the algorithm, because on that page there is an extended and updated write-up on this method in addition to a copy of the original paper. That write-up is more useful as an introduction to the method, but obviously has different equation numbers etc. thus it is not a good link for the `_asses_dimension` method, which explicitly refers to "equation (31) from the paper".

glemaitre commented on sklearn/decomposition/_pca.py:
```diff
@@ -53,7 +53,8 @@ def _assess_dimension(spectrum, rank, n_samples):
     -----
     This implements the method of `Thomas P. Minka:
-    Automatic Choice of Dimensionality for PCA. NIPS 2000: 598-604`
+    Automatic Choice of Dimensionality for PCA. NIPS 2000: 598-604
```
I think that we could remove the note and add the sentence in the long discussion on the top of the docstring and add a proper `References` section.

hamogu replied:
done, but for consistency with how the References section looks in the main PCA doctring, I left the sentence also in the Referneces section.

glemaitre commented on sklearn/decomposition/_pca.py:
```diff
@@ -273,6 +274,7 @@ class PCA(_BasePCA):
     ----------
     For n_components == 'mle', this class uses the method of *Minka, T. P.
     "Automatic choice of dimensionality for PCA". In NIPS, pp. 598-604*
+    See https://tminka.github.io/papers/pca/
```
Could you create an hyperlink here such that the part from `Minka ... 604` would be clickable on the HTML.

hamogu committed:
Added ULRs for all references in the PCA docstring

applied comments from review, homogenized layout for all references in the main PCA docstring

hamogu commented:
Unfortunately, I know find myself wit ha linting error, that I don't understand. Any advice?
As a package maintainer for other packages, I understand how annoying it is when the author of a PR does not follow through, but I've spend about an hour now trying to lix my local linter to understand where this message comes from with no success so far. I wanted to contribute a quick improvement, but don't have enough time to delve any deeper into that right now.

Do you see the problem?

jeremiedbb commented on sklearn/decomposition/_pca.py:
```diff
@@ -271,26 +273,30 @@ class PCA(_BasePCA):
-    "Automatic choice of dimensionality for PCA". In NIPS, pp. 598-604*
+    For n_components == 'mle', this class uses the method from:
+    `Minka, T. P.. "Automatic choice of dimensionality for PCA".
+    In NIPS, pp. 598-604 <https://tminka.github.io/papers/pca/>`_
```
Why not giving the same link as in the ``_assess_dimension` function (https://proceedings.neurips.cc/paper/2000/file/7503cfacd12053d309b6bed5c89de212-Paper.pdf) ? It links directly to a pdf.

hamogu replied:
The link I give here links to Minka's page which has the pdf of the original publication, but also an updated and expanded version that explains the algorithm in more detail. I found it much easier to understand. People reading this general docstring may want to understand the algorthim in details. On the other hand, in the `_assess_dimension` there is an explicit reference to "equation 31", so one needs to look at the exact paper.

jeremiedbb replied:
Thanks for the clarification. Then I think it's nicer to the reader to link directly to the pdf of the expanded version (https://tminka.github.io/papers/pca/minka-pca.pdf)

hamogu replied:
Done.

hamogu commented on sklearn/decomposition/_pca.py:
```diff
@@ -271,26 +273,30 @@ class PCA(_BasePCA):
-    "Automatic choice of dimensionality for PCA". In NIPS, pp. 598-604*
+    For n_components == 'mle', this class uses the method from:
+    `Minka, T. P.. "Automatic choice of dimensionality for PCA".
+    In NIPS, pp. 598-604 <https://tminka.github.io/papers/pca/>`_
```
```suggestion
    In NIPS, pp. 598-604 <https://tminka.github.io/papers/pca/minka-pca.pdf>`_
```

hamogu committed:
Update reference to link to exanded version directly

hamogu commented:
linter failed with 
```
error: pathspec 'pr/19207/merge' did not match any file(s) known to git
Could not fetch merge commit.
There may be conflicts in merging PR #19207 with master.
```
I rebased on current main, but that did not change the situation. Probably related to `Base automatically changed from master to main 1 hour ago`?

jeremiedbb commented:
Yes it should be fixed by #19237

jeremiedbb reviewed (approved):
LGTM

glemaitre commented:
Thanks @hamogu
